### PR TITLE
fix: race condition on `newPage`

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes https://github.com/apify/apify-js/issues/1126

### CODE REVIEW

File: test/browser-pool.test.ts [hunk 1]
```diff
@@ -199,6 +199,25 @@ describe('BrowserPool', () => {
             expect(browserPool['_launchBrowser']).toBeCalledTimes(3);
         });
 
+        test('should allow max pages per browser - no race condition', async () => {
+            browserPool.maxOpenPagesPerBrowser = 1;
+            // @ts-expect-error Private function
+            jest.spyOn(browserPool!, '_launchBrowser');
```
Comment (B4nan) [inline test/browser-pool.test.ts:205]: looks a bit weird that
1. you use `!` here but not elsewhere in the file (so probably not needed)
2. you dont have the expect error comment on the last line of this test, where you also access `_launchBrowser` (so i'd expect either both to fail on non to fail)

**Review by B4nan — APPROVED** [review]
looking good, just one question

File: test/browser-pool.test.ts (hunk 1 above)
Comment (szmarczak) [reply test/browser-pool.test.ts]: 1. leftover from wip (fixed, thanks)
2. it's accessed through brackets - with the current ts configuration you can access private properties with the brackets operator

**Review by mnmkng — APPROVED** [review]

**Comment by B4nan** [comment]
i guess we'll need a version bump too

**Pull request merged**